ion/decimal.go

```diff
@@ -274,10 +272,9 @@ func (d *Decimal) trunc() (int64, error) {
 	if err != nil {
 		return 0, err
 	}
-	d = ud
```

Comment:
I think we need this.
```ud``` is different from the original ```d```.  I think when we change the scale, we need to keep ```d``` with its updated version in case another operation is called on ```d```

Reply:
Consider `d.checkToUpscale()`. 
In the old code, it reassigns d with `d = d.upscale(0)`. However I discovered that this change of state is not reflected in the calling function `d.trunc()` which must be why `d.checkToUpscale()` returns the new decimal. (Yes I'm confused by this as well since I expect a pointer receiver to change state.)